common/persistence/client/factory.go

```diff
@@ -169,6 +171,9 @@ func (f *factoryImpl) NewShardManager() (p.ShardManager, error) {
 		return nil, err
 	}
 	result := p.NewShardManager(store)
+	if errorRate := f.config.ErrorInjectionRate(); errorRate != 0 {
+		result = p.NewShardPersistenceErrorInjectionClient(result, errorRate, f.logger)
```

Comment:
How about passing in `f.config.ErrorInjectionRate`? since it's dynamic config we should take the advantage. Otherwise may just use static config then.

Reply:
I was worry about memory issue by querying dynamic config value too often. I retied once on staging by reading a dynamicconfig value once per second for each shard and it resulted some memory issues (GC can't keep up). For persistence client I think the rate will be even higher so would like to keep it safe.

Also I think we will never use this error injection client in production so the value won't be changed, and we can have one less wrapper around persistence manger. The only place it will be used is in our testing environment, where I think restarting the cluster is acceptable.

Reply:
Makes sense to me. I am fine with that.